### DISCUSSION TITLE
Foreign character problem

### DISCUSSION THREAD

**Discussion by saschke** [fo-dicom/fo-dicom#1305] [Q&A]
Hi all

I have a foreign character Issue (Umlaut problem) that I haven't been able to solve after hours so I try here

I use FO-Dicom 4.0.8 in an application written in .NET Framework 4.72 and use the sample code for the work list SCU to access a Work list provider (a worker in .NET 5), also FO-Dicom.
The SCP works fine with my GE Ultrasound device and the names I get have correct umlauts.

When I use the worklist user sample code I get worklist item encoded in ASCII (and question marks instead of the umlauts) and no matter what I try (adding Encoding to the request and even adding a fix encoding on the server side) I don't manage to get correct umlauts.

So here my question: how can I force the server to send me the correct encoding? I thought by adding the encoding to the request

            dataset.AddOrUpdate(new DicomTag(0x8, 0x5), Encoding.GetEncoding("iso-8859-1"));

or / and by adding it to the item of the request

            dicomCfindRequest.Dataset.Add<string>(DicomTag.PatientName, Encoding.GetEncoding("iso-8859-1"), string.Empty);

but that doesn't change anything.

There must be a way because it works on the GE. What am I doing wrong?

Thanks for any help! I hope I made myself clear.

-Schlomo

**Comment by gofal** [comment]
as far as I know you cannot enfoce the encoding of the answer. it's the servers decission which encoding the result has. Maybe there are some implementations out there that they take the encoding that was used in the request, but you cannot expect every implementation to behave that way. the one who creates a Dataset may decide which encoding to use. because the encoding may depend on the content

**Reply by saschke** [reply]
I got it to work

I added:      resultDataset.AddOrUpdate(new DicomTag(0x8, 0x5), "ISO_IR 100");

to the WorkListHandler.cs of the demo-code (s. below)

           //  Parsing result 
            foreach (var result in results)
            {
                var resultingSPS = new DicomDataset();
                var resultDataset = new DicomDataset();
                var resultingSPSSequence = new DicomSequence(DicomTag.ScheduledProcedureStepSequence, resultingSPS);

                **resultDataset.AddOrUpdate(new DicomTag(0x8, 0x5), "ISO_IR 100");**

and I changed every AddIfExistsInRequest(...) that could have special characters to include the correct encoding

private static readonly Encoding EncodingForNames = Encoding.Latin1; 

AddIfExistsInRequest(resultDataset, request, DicomTag.PatientName, EncodingForNames, result.Surname + "^" + result.Forename + "^^" + result.Title); 

This works for the WorkList SCU as well as the ultrasound devices

Thanks

-Schlomo
